crates/ruff_python_ast/src/helpers.rs

```diff
@@ -1746,4 +1778,118 @@ y = 2
 
         Ok(())
     }
+
+    #[test]
+    fn any_over_stmt_type_alias() {
```

Comment:
This test is a bit contrived but I'd appreciate any general Rust feedback as I'm using it for practice!

Reply:
I think it's fine. Or let's say, it's what the API allows you to do. You could have considered changing the `Fn` to `FnMut` so that you don't need to use `RefCell` (I guess it was a good learning opportunity ;)). 

I'm normally too lazy to bother with building the tree manually and instead parse the source code. Now, using the source code makes asserting a bit annoying because you don't have access to the node instances. That's where you could collect `NodeKind`s instead (you can get it by calling `AnyNodeRef::from(expr).kind()`).  However, `NodeKind` only works if you never have to distinguish between any two nodes of the same kind. These are some tests that I added that use `NodeKind` https://github.com/astral-sh/ruff/blob/5d41c832ade7f5fcde5b535a8fb9784ec199489f/crates/ruff_python_ast/src/visitor/preorder.rs#L955-L1039


That said. I think what you have is good. The only thing I'm not a 100% sure of is if I would bother refactoring this test if I ever end up breaking it (thinking about added value / maintenance cost)

Reply:
Thanks for taking a look!

> You could have considered changing the Fn to FnMut

It seems wrong to change the API just for testing.

> I'm not a 100% sure of is if I would bother refactoring this test if I ever end up breaking it 

Definitely. That's kind of what I was getting at with it being a bit contrived.

> I'm normally too lazy to bother with building the tree manually and instead parse the source code.

Ah that's very wise. I feel like that's much more maintainable. I'll look into doing that.

Reply:
Eh this is kind of hard to do. I'd rather address it in a follow-up or next time someone wants to add a test here.